### ISSUE TITLE
Improve cleaning up of preview envs ( jx gc previews )

### ISSUE THREAD

**Issue by mrpity** [jenkins-x/jx#1336]
### Summary 
Cmd: *jx gc previews* doesn't delete config map that is created by HELM after preview installation.

### Steps to reproduce the behavior
- create preview env
- close PR
- wait for cronjob execution or run manually *jx gc previews*

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 1.3.109
jenkins x platform 0.0.1881
kubernetes cluster v1.10.5
kubectl            v1.11.0
helm client        v2.10.0-rc.1+gaa98e7e
helm server        v2.10.0-rc.1+gaa98e7e
git                git version 2.11.0
```

### Kubernetes cluster

What kind of Kubernetes cluster are you using & how did you create it?

 - kubernetes on-premise
 - kubespray

### Operating system / Environment
 - Ubuntu 16.04.4 LTS

### Expected behavior
 I expect that configmaps for all preview envs should be deleted after execution *jx gc previews*

### Actual behavior

```
# jx get preview
PULL REQUEST                                           NAMESPACE                         APPLICATION
https://bitbucket.org/dkhodakivsky/jx_test_repo/pull/8 jx-dkhodakivsky-jx-test-repo-pr-8 http://jx-test-repo.jx-dkhodakivsky-jx-test-repo-pr-8.isk8s.whirl.sg

# jx gc preview
Deleted environment dkhodakivsky-jx-test-repo-pr-8

# kubectl get configmap --all-namespaces | grep dkho
kube-system     jx-dkhodakivsky-jx-test-repo-pr-8.v1   1         1h

# helm list | grep dkho
jx-dkhodakivsky-jx-test-repo-pr-8    1           Mon Jul 23 12:24:19 2018    DEPLOYED    preview-0.0.0-SNAPSHOT-PR-8-1                   jx-dkhodakivsky-jx-test-repo-pr-8

```

**Comment by rawlingsj** [comment]
Looks like this is because we're not doing a `helm del --purge foo` which should clean up the configmap used by tiller https://github.com/helm/helm/issues/2332

**Comment by polothy** [comment]
We are seeing this as well.  Our pull requests create preview environments.  After the PR is merged, the environment (k8s namespace) is cleaned up but doing a `helm ls` still shows the release/deployment to the preview environment.

**Comment by polothy** [comment]
Digging through the code, looks like this needs updating perhaps? `DeleteEnvOptions.deleteEnviroment`

Unsure how to go about the helm deletion since it could be helm3, tillerless or prow (maybe prow does not matter here).  Is there already a generic way to do this?

**Comment by rawlingsj** [comment]
Yes exactly, I just got to the same function https://github.com/jenkins-x/jx/blob/master/pkg/jx/cmd/delete_env.go#L131-L133

We should delete the helm release here too

Do you fancy having a pop at a pull request?

/kind bug
/priority important-soon

**Comment by polothy** [comment]
I wouldn't mind writing one, but I wasn't sure exactly what I would need to call.  Perhaps the only use case we care about is if we actually are using tiller and all the other use cases do not need extra cleanup?

**Comment by rawlingsj** [comment]
Yes I think that's correct, it's only when tiller is being used.  I think there's a helper function to tell if we have a tillerless install, maybe something like this https://github.com/jenkins-x/jx/blob/master/pkg/jx/cmd/common_helm.go#L241

**Comment by polothy** [comment]
Thanks for the pointers, this is where I'm at:
```diff
Index: pkg/jx/cmd/delete_env.go
===================================================================
--- pkg/jx/cmd/delete_env.go	(revision f6f1699f4cec8f2f0d07b7dc7a1a7b1e4725ede4)
+++ pkg/jx/cmd/delete_env.go	(date 1545162039000)
@@ -129,6 +129,17 @@
 	}
 	kind := env.Spec.Kind
 	if o.DeleteNamespace || !kind.IsPermanent() {
+		helmBin, noTiller, helmTemplate, err := o.TeamHelmBin()
+		if err != nil {
+			return err
+		}
+		// Only run helm delete when using tiller and not template or helm3.
+		if !noTiller && !helmTemplate && helmBin == defaultHelmBin {
+			err := o.Helm().DeleteRelease(envNs, envNs, true)
+			if err != nil {
+				return err
+			}
+		}
 		return o.KubeClientCached.CoreV1().Namespaces().Delete(envNs, &metav1.DeleteOptions{})
 	}
 	log.Infof("To delete the associated namespace %s for environment %s then please run this command\n", name, envNs)
```

The problem is that this line feels very wrong: `err := o.Helm().DeleteRelease(envNs, envNs, true)`.  While in _most_ cases, the release name is the namespace name, I'm sure at some point that will not be the case.  I was going to use `HelmCLI.StatusReleases` to get all the releases in a given namespace, then delete each one, but that method doesn't appear to respect the namespace argument.  Any suggestions?
